src/mapgen/mapgen_v6.cpp

```diff
@@ -132,6 +132,39 @@ MapgenV6::MapgenV6(MapgenV6Params *params, EmergeManager *emerge)
 		c_stair_cobble = c_cobble;
 	if (c_stair_desert_stone == CONTENT_IGNORE)
 		c_stair_desert_stone = c_desert_stone;
+
+	if (c_stone == CONTENT_IGNORE)
+		errorstream << "Mapgen v6: Mapgen alias 'mapgen_stone' is invalid!" << std::endl;
```

Comment:
This is a lot of duplicate code, and there exists an additional `getId` method which returns a success boolean:
http://github.com/minetest/minetest/blob/3d6b55d3e9b91e6afdbfdcea0268dfb88c69c1c2/src/nodedef.h#L493-L507
http://github.com/minetest/minetest/blob/3d6b55d3e9b91e6afdbfdcea0268dfb88c69c1c2/src/nodedef.cpp#L1007-L1023

Maybe it is possible to use [lambda expressions](https://en.cppreference.com/w/cpp/language/lambda) to shorten the code, and set the node to air if the mapgen alias is missing.
```c++
[…]
	auto getIdValid = [&](const std::string &name)->content_t
	{
		content_t id = CONTENT_AIR;
		if (!ndef->getId(name, id))
			errorstream << "Mapgen v6: Mapgen alias '" << name <<
				"' is invalid!" << std::endl;
		return id;
	}

	c_stone           = getIdValid("mapgen_stone");
	c_dirt            = getIdValid("mapgen_dirt");
[…]
```

Reply:
I am neutral about using a lambda expression, but it should only be used for mgv6.
Non-mgv6 mapgens only need 3 aliases checked, so in that case simple repeated conditionals will be fewer lines and easier to understand.

Also, why does the lambda expression example above set content ID to CONTENT_IGNORE? If the alias is missing the content ID of the mapgen node will already be CONTENT_IGNORE.
Please only add code to trigger an error, nothing else.

To be honest i am still confused by HybridDog's code example.
I am not sure `ndef->getId(name, id)` is needed because `ndef->getId(name)` looks just like the code example above, see
https://github.com/minetest/minetest/blob/3d6b55d3e9b91e6afdbfdcea0268dfb88c69c1c2/src/nodedef.cpp#L1018-L1023

Reply:
> If the alias is missing the content ID of the mapgen node will already be CONTENT_IGNORE.

No, the getId method with two arguments does not set it to ignore automatically.

> I am not sure `ndef->getId(name, id)` is needed […]

The getId method with one argument is just a wrapper for the other one. 
Instead of adding a lambda expression, an additional wrapper could be added to NodeDefManager, which can be used in all the mapgen files.

Reply:
But lines 95-107 have already set the ID to ignore if the alias is missing ... ?

Is your code example meant to be inserted at line 136 or line 95? I assumed 136.
If you mean line 95 the code is wrong, because checking for a missing alias must be done after the fallbacks are applied, not before.
For example if the stone alias is present but the desert_stone alias is missing, your code will trigger an error even though afterwards c_desert_stone falls back to c_stone, which is fine and should not trigger an error.

If you mean line 136 then the code is also wrong for reasons already explained.

I will close this comment for now but feel free to rethink, fix and resubmit a code suggestion.

Reply:
> If you mean line 95 the code is wrong […]

I thought about inserting it to line 95 and using it only for the IDs which don't have an alias.